File: Php/weather.php
<?php
$link = 'https://api.open-meteo.com/v1/forecast?latitude=44.76&longitude=19.22&hourly=temperature_2m,snowfall&timezone=Europe%2FBerlin&start_date=2022-12-17&end_date=2022-12-21';
function getWeather($url){
    $json = file_get_contents($url);
    $obj = json_decode($json);
    $temperature = $obj->hourly->temperature_2m;
    $time = $obj->hourly->time;
    $snowfall = $obj->hourly->snowfall;
    $str = "TIME;TEMPERATURE;SNOWFALL\n";

    for($i=0; $i < count($time); $i++) {
        $str .= $time[$i] .";". $temperature[$i]. ";". (($snowfall[$i]==0) ? "false" : $snowfall[$i]) .";\n";
    }
    return $str;
}
// print_r($str);
$str=getWeather($link);
$f=fopen("Weather.scv","w") or die("Unable to open file.");
fwrite($f,$str);
fclose($f);

function hasWeatherChanged(){
    $str2 = '';
    $f=fopen("Weather.scv","r") or die("Unable to open file.");
// echo fread($f,filesize("Weather.csv"));
    while(!feof($f)){
        $str2 .= fgets($f);
    }
    fclose($f);
    $format1 = date("Y-m-d");
    $format2 = date("Y-m-").date("d")+5;
    $link2 ='https://api.open-meteo.com/v1/forecast?latitude=44.76&longitude=19.22&hourly=temperature_2m,snowfall&timezone=Europe%2FBerlin&start_date='.$format1.'&end_date='.$format2;
    $str1 = getWeather($link2);
    if($str2 != $str1){
        $f=fopen("Weather.scv","w") or die("Unable to open file.");
        fwrite($f,$str1);
        fclose($f);
        echo("Weather changed.\n");
    }else{
        echo("Nothing changes.\n");
    }
}
while(true){
    hasWeatherChanged();
        sleep(5);
}

?>

File: Python/weather.py
#!/usr/bin/env python3
import requests
import json
import datetime
from apscheduler.schedulers.background import BlockingScheduler

link = "https://api.open-meteo.com/v1/forecast?latitude=44.76&longitude=19.22&hourly=temperature_2m,snowfall&timezone=Europe%2FBerlin&start_date=2022-12-17&end_date=2022-12-21"
# f = urllib.urlopen(link)
def getWeather(link):
    f = requests.get(link)
    # myfile = f.read()
    data = json.loads(f.text)
    time = data["hourly"]["time"]
    temperature = data["hourly"]["temperature_2m"]
    snowfall = data["hourly"]["snowfall"]
    string = "TIME;TEMPERATURE;SNOWFALL\n"
    for i, x, y in zip(time, temperature, snowfall):
        string += i + ";" + str(x) + ";" + "false;\n" if y == 0 else str(y) + ";\n"
    return string


string = getWeather(link)
file1 = open("Weather.csv", "w")
file1.write(string)
file1.close()


def hasWeatherChanges():
    date = datetime.datetime.now()
    format1 = str(date.year) + "-" + str(date.month) + "-" + str(date.day)
    format2 = str(date.year) + "-" + str(date.month) + "-" + str(date.day + 5)

    link2 = f"https://api.open-meteo.com/v1/forecast?latitude=44.76&longitude=19.22&hourly=temperature_2m,snowfall&timezone=Europe%2FBerlin&start_date={format1}&end_date={format2}"

    string2 = getWeather(link2)
    string1 = open("Weather.csv", "r").read()
    if string1 != string2:
        f = open("Weather.csv", "w")
        f.write(string2)
        f.close()
        print("Weather.csv are changed.")
    else:
        print("Nothing changes.")


sched = BlockingScheduler()
sched.add_job(
    hasWeatherChanges, "interval", seconds=60
)  # will do the print_t work for every 60 seconds

sched.start()
